### ISSUE TITLE
Devastation Evoker

### ISSUE THREAD

**Issue by chuwix** [Twintop/TwintopInsanityBar#259]
Hi, I love your addon.

Any chance for supporting visualization of Evoker's Essences?

**Comment by Twintop** [comment]
Glad to hear you find it useful!

tldr: Yes, just not sure when. Hopefully before Dragonflight launch but not for prepatch.

Longer:
Yes, absolutely something I want to do. I'm behind on getting existing specs updated before prepatch, though, so that's going to be my focus. Once they are all back in a working state (5 still to be done, with a 6th needing options menu updates) and I have refreshed all the bar text variables, then I'll start working on Evoker. I'm hoping to have something working before they are playable on live but it could easily slip to (or past) Dragonflight release. This is a very busy time for me (outside of WoW) so finding time to get things added is a challenge.

There are a couple of technical hurdles I need to overcome with Evoker that might cause delays, only because the bar doesn't have any specs that use similar tech yet. Essence is similar to DK Runes (or, maybe even, some Warlock Soulshards) in that they're a secondary resource that (mostly) renews on a timer or can have "partial" values. I'm planning on using the existing Combo Points / Chi UX for these, but those don't have any code written to handle these partial updates, instead they are basically either on/off. That will probably be the biggest time sink and delay.

If there are any things you would find helpful to have tracked or included as options/cues/etc. for Devastation or Preservation (or any other specs, currently supported or not), please don't hesitate to let me know!

**Comment by chuwix** [comment]
Thank you very much for the detailed and quick answer.
Would you consider starting with an MVP that would quickly re-use as much tech as possible to make Essences work by the phase 2 prepatch (Evoker release) and iterate upon that at a later point?

**Comment by Twintop** [comment]
Yes, that will be the first phase of adding either Devastation or Preservation. What I've done in the past for specs that aren't quite ready for widespread use but have some things working is I add them as an opt-in from the Options menu, e.g. for live:

![image](https://user-images.githubusercontent.com/1485707/196571492-9066f1d7-6f67-4eb3-b846-15753a8a3595.png)

Something similar makes sense and would be noted in any release notes for that version of the bar.

**Comment by Twintop** [comment]
Prevervation being tracked here #280. I'll track Devastation in this issue.

**Comment by Twintop** [comment]
Well I have some UI rendering now. 
![image](https://user-images.githubusercontent.com/1485707/215261773-c2201cb7-07f5-4626-9b8e-bf9fc15059c0.png)

**Comment by Twintop** [comment]
@Maroslav A little late and definitely minimal, but this is available as an experimental feature as of today's release of 10.0.5.2-release.

**Comment by Twintop** [comment]
Outstanding work to get this out of minimal/experimental and in to general availability by default (WIP, list not final):

- Essence Notifications
- - [x] Ruby/Azure Essence Burst + Essence Attunement
- - - [x] Bar text variables, `$ebTime` and `$ebStacks`
- - - [x] Bar border color change (1 or 2?)
- - - [x] Audio cue

**Comment by Twintop** [comment]
There's not a lot more to be added, unless anyone has specific requests. Closing this to clean the backlog up to prepare for TWW.
